features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2

```diff
@@ -2072,7 +2072,8 @@
         {% if authentication.jit_provisioning.allow_idp_login is defined %}
             <AllowLoginToIDP>{{authentication.jit_provisioning.allow_idp_login}}</AllowLoginToIDP>
         {% endif %}
-        <AllowAssociatingToExistingUser>{{authentication.jit_provisioning.associating_to_existing_user}}</AllowAssociatingToExistingUser>
+        <AllowAssociatingToExistingUser>{{authentication.jit_provisioning.associating_to_existing_user | default(true)}}</AllowAssociatingToExistingUser>
```

Comment:
The default value jit_provisioning.associating_to_existing_user of is  set to `false` in our default.json file [1] with [2], so setting it to true by default in here won't take effect AFAIU. @RushanNanayakkara can you please give the context of the current behaviour and the initial purpose of setting this configuration to false by default?

[1] - https://github.com/wso2/carbon-identity-framework/blob/master/features/authentication-framework/org.wso2.carbon.identity.application.authentication.framework.server.feature/resources/org.wso2.carbon.identity.application.authentication.framework.server.feature.default.json#L273
[2] - https://github.com/wso2/carbon-identity-framework/pull/5468

Reply:
This value was set to false by default due to a critical security vulnerability. Why do we need to set this to true? 
As per the below config, it seems by default we will not be retaining the email domain. This is a vulnerable configuration and should not be the packs default configuration.